src/Components/test/E2ETest/Tests/SvgTest.cs

```diff
@@ -0,0 +1,143 @@
+using BasicTestApp;
+using BasicTestApp.RouterTest;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.E2ETesting;
+using Microsoft.AspNetCore.Testing;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Interactions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETest.Tests
+{
+    public class SvgTest : ServerTestBase<ToggleExecutionModeServerFixture<Program>>
+    {
+        public SvgTest(
+            BrowserFixture browserFixture,
+            ToggleExecutionModeServerFixture<Program> serverFixture,
+            ITestOutputHelper output)
+            : base(browserFixture, serverFixture, output)
+        {
+        }
+
+        protected override void InitializeAsyncCore()
+        {
+            Navigate(ServerPathBase, noReload: _serverFixture.ExecutionMode == ExecutionMode.Client);
+        }
+
+        [Fact]
+        public void CanRenderSvgWithCorrectNamespace()
+        {
+            var appElement = Browser.MountTestComponent<SvgComponent>();
+
+            var svgElement = appElement.FindElement(By.Id("svg-with-callback"));
+            Assert.NotNull(svgElement);
+
+            var svgCircleElement = svgElement.FindElement(By.XPath("//*[local-name()='circle' and namespace-uri()='http://www.w3.org/2000/svg']"));
+            Assert.NotNull(svgCircleElement);
+            Assert.Equal("10", svgCircleElement.GetAttribute("r"));
+
+            appElement.FindElement(By.TagName("button")).Click();
+            Browser.Equal("20", () => svgCircleElement.GetAttribute("r"));
+        }
+
+        [Fact]
+        public void CanRenderSvgChildComponentWithCorrectNamespace()
+        {
+            var appElement = Browser.MountTestComponent<SvgComponent>();
+
+            var svgElement = appElement.FindElement(By.Id("svg-with-child-component"));
+            Assert.NotNull(svgElement);
+
+            var svgCircleElement = svgElement.FindElement(By.XPath("//*[local-name()='circle' and namespace-uri()='http://www.w3.org/2000/svg']"));
+            Assert.NotNull(svgCircleElement);
+        }
+
+        [Fact(Skip="https://github.com/dotnet/aspnetcore/issues/18271")]
+        public void CanRenderVariablesInForeignObject()
+        {
+            var appElement = Browser.MountTestComponent<SvgComponent>();
+
+            var svgElement = appElement.FindElement(By.Id("svg-with-foreign-object"));
+            Assert.NotNull(svgElement);
+
+            Func<IEnumerable<IWebElement>> strongElement =
+                () => appElement.FindElements(By.TagName("strong"));
+
+            Browser.Collection<IWebElement>(strongElement,
+                e => Assert.Equal("thestringfoo", e.Text),
+                e => Assert.Equal("thestringbar", e.Text));
+        }
+
+        [Fact(Skip="https://github.com/dotnet/aspnetcore/issues/18271")]
+        public void CanRenderSvgWithLink()
```

Comment:
Is your approach here to just leave the tests in but skipped and then fix them and turn them on?

Reply:
Correct -- the plan was to go through all the issues and create test scenarios out of them so we have a sense of what's broken. Then we go through and address the most impactful ones.